### PR TITLE
Add spm_current_partition_id() API and tests for it

### PR DESCRIPTION
**Pull request by romkuz01**
Add psa_current_partition_id() API function and two tests for it (one for NSPE call and another for SPE).

---
This change is [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/50)

### CODE REVIEW

File: spm/spm_client.c [hunk 1]
```diff
@@ -283,6 +283,13 @@ void psa_close(psa_handle_t handle)
     spm_rot_service_queue_enqueue(channel->dst_rot_service, channel);
 }
 
+int32_t psa_current_partition_id(void)
```
Comment (orenc17) [inline spm/spm_client.c]: please make it inline function

File: spm/spm_client.h [hunk 2]
```diff
@@ -99,6 +99,13 @@ psa_error_t psa_call(
  */
 void psa_close(psa_handle_t handle);
 
+/**
+ * Return to the calling partition its ID.
+ *
+ * @return current paretition ID or PSA_NSPE_IDENTIFIER if called by NSPE.
```
Comment (danny4478) [inline spm/spm_client.h]: Typo: "paretition"

File: spm/spm_client.h [hunk 3]
```diff
@@ -99,6 +99,13 @@ psa_error_t psa_call(
  */
 void psa_close(psa_handle_t handle);
 
+/**
+ * Return to the calling partition its ID.
```
Comment (danny4478) [inline spm/spm_client.h]: I would change to "Return caller's partition ID"

File: spm/spm_client.h (hunk 2 above)
Comment (romkuz01) [reply spm/spm_client.h]: Fixed.

File: spm/spm_client.h (hunk 3 above)
Comment (romkuz01) [reply spm/spm_client.h]: Changed.

File: spm/spm_client.c (hunk 1 above)
Comment (romkuz01) [reply spm/spm_client.c]: Added "inline" keyword.

**Comment by danny4478** [comment]
Hi,

Can you please add the new tests to our excel?
Note there are 2 tabs for this excel for Positive and Negative tests.

https://armh.sharepoint.com/:x:/r/sites/iot-project-o365/_layouts/15/doc2.aspx?sourcedoc=%7B0f4f01b1-d3c4-4bd1-b584-0972a1d3acbb%7D&action=edit&activeCell=%27IPC%20Negative%20Tests%27!A2&wdInitialSession=6c007101-91f2-4999-a73a-238780180c59&wdRldC=1

**Pull request closed without merging**